### PR TITLE
Fix -Wextra-semi warnings

### PR DESCRIPTION
**Pull request by Kangz**
PTAL and no worries if you think this is not worth fixing.

### CODE REVIEW

**Comment by Kangz** [comment]
Up! @elmindreda PTAL (I'll fix the merge issue with README.md if this looks ok)

**Comment by elmindreda** [comment]
LGTM. Would you mind doing the other ones as well? `cocoa_joystick.h`, `null_joystick.h`, `null_platform.h`, `win32_joystick.h` has various other (bad) approaches to this.

**Comment by Kangz** [comment]
Done, though I haven't tested compilation on all platform. AFAIK CI covers all configurations?

**Comment by Kangz** [comment]
Had to add an `int dummy` in the struct otherwise MSVC produces this error: `error C2016: C requires that a struct or union has at least one member`

**Comment by Kangz** [comment]
Bump! I ran into this again today.

File: src/null_platform.h [hunk 1]
```diff
@@ -29,13 +29,13 @@
...
-#define _GLFW_PLATFORM_LIBRARY_CONTEXT_STATE
-#define _GLFW_EGL_CONTEXT_STATE
-#define _GLFW_EGL_LIBRARY_CONTEXT_STATE
+#define _GLFW_PLATFORM_CONTEXT_STATE         struct { int dummy; }
+#define _GLFW_PLATFORM_MONITOR_STATE         struct { int dummy; }
+#define _GLFW_PLATFORM_CURSOR_STATE          struct { int dummy; }
+#define _GLFW_PLATFORM_LIBRARY_WINDOW_STATE  struct { int dummy; }
+#define _GLFW_PLATFORM_LIBRARY_CONTEXT_STATE struct { int dummy; }
+#define _GLFW_EGL_CONTEXT_STATE              struct { int dummy; }
+#define _GLFW_EGL_LIBRARY_CONTEXT_STATE      struct { int dummy; }
```
Comment (elmindreda) [inline src/null_platform.h]: This causes name conflict errors because more than one of these goes into the library struct.

Comment (Kangz) [reply src/null_platform.h]: Done

**Comment by Kangz** [comment]
@elmindreda PTAL again!

**Comment by Kangz** [comment]
Thanks!

**Comment by Luflosi** [comment]
Wouldn't it have been nicer to just put the semicolons where the `#define`s are instead of doing the IMO ugly workaround with the `dummyContext`? I can make a PR to change this if it will be accepted.

**Comment by elmindreda** [comment]
@Luflosi Agreed. I would accept this PR, only I'm hoping to merge runtime platform selection soon and that will remove these altogether.

**Pull request closed without merging**